proxy/src/main/java/org/apache/rocketmq/proxy/processor/ProducerProcessor.java

```diff
@@ -102,11 +108,19 @@ public CompletableFuture<List<SendResult>> sendMessage(ProxyContext ctx, QueueSe
                         if (SendStatus.SEND_OK.equals(sendResult.getSendStatus()) &&
                             tranType == MessageSysFlag.TRANSACTION_PREPARED_TYPE &&
                             StringUtils.isNotBlank(sendResult.getTransactionId())) {
-                            fillTransactionData(ctx, producerGroup, messageQueue, sendResult, messageList);
+                            fillTransactionData(ctx, producerGroup, finalMessageQueue, sendResult, messageList);
                         }
                     }
                     return sendResultList;
-                }, this.executor);
+                }, this.executor)
+                    .whenComplete((result, exception) -> {
+                        endTimestamp.set(System.currentTimeMillis());
```

Comment:
There is no need to use AtomicLong for `endTimestamp`, just a local variable would be fine